config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_vertexaiindexes.vertexai.cnrm.cloud.google.com.yaml

```diff
@@ -222,60 +222,311 @@ spec:
                       type: string
                   type: object
                 type: array
-              createTime:
-                description: The timestamp of when the Index was created in RFC3339
-                  UTC "Zulu" format, with nanosecond resolution and up to nine fractional
-                  digits.
+              observedGeneration:
+                description: ObservedGeneration is the generation of the resource
+                  that was most recently observed by the Config Connector controller.
+                  If this is equal to metadata.generation, then that means that the
+                  current reported status reflects the most recent desired state of
+                  the resource.
+                type: integer
+              observedState:
+                description: The observed state of the underlying GCP resource.
+                properties:
+                  createTime:
+                    description: The timestamp of when the Index was created in RFC3339
+                      UTC "Zulu" format, with nanosecond resolution and up to nine
+                      fractional digits.
+                    type: string
+                  deployedIndexes:
```

Comment:
Looks like this resource has a similar issue about lacking clarification about the deployment use case. From the public doc, it looks like another resource, indexEndpoint, is needed to deploy the index: https://cloud.google.com/vertex-ai/docs/vector-search/deploy-index-vpc#create_index_endpoint-drest But it's unclear how we plan to support it in KCC.

It's also a bit concerning that `deployedIndexes` doesn't seem to be generated by the creation of the VertexAIIndex, but by [deployIndex()](https://cloud.google.com/vertex-ai/docs/reference/rest/v1beta1/projects.locations.indexEndpoints/deployIndex), an endpoint under a different resource.

We may want to get some clarification about the use case of this resource from both the user and the API team before we determine the interface of this resource.

Reply:
So it looks like Terraform has "index" and "index_endpoint", but to deploy an "index" to an "index_endpoint", users need to call GCP API `POST https://{service-endpoint}/v1beta1/{indexEndpoint}:deployIndex`. To make it safe, maybe we should ignore the field for now.

Reply:
@yuwenma fyi this is the other tricky API design issue.

IIUC,
* Users can create the index.
* Users can deploy the index to the index endpoint.
* Users can deploy the index to the index endpoint (not sure whether it's the same index endpoint or not) multiple times.
* Deployed index will be represented as an output-only `deployedIndexes` list in the index resource.
* The index resource can't be deleted unless all the deployed indexes has been undeployed.

I think it is a reasonabe workaround to ignore `deployedIndexes` for now, though @jingyih could you ensure deletion won't stuck in weird state / error message is clear if there are deployed indexes but not represented by the index resource?

Reply:
Ignored `deployedIndexes` for now. I will try to test this manually.